src/test/scala/ch/epfl/bluebrain/nexus/kg/storage/ExternalDiskStorageOperationsSpec.scala

```diff
@@ -0,0 +1,104 @@
+package ch.epfl.bluebrain.nexus.kg.storage
+
+import java.nio.file.Paths
+
+import akka.http.scaladsl.model.ContentTypes._
+import akka.http.scaladsl.model.Uri
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import cats.effect.IO
+import ch.epfl.bluebrain.nexus.commons.test.io.IOEitherValues
+import ch.epfl.bluebrain.nexus.commons.test.{ActorSystemFixture, Randomness, Resources}
+import ch.epfl.bluebrain.nexus.iam.client.types.{AuthToken, Permission}
+import ch.epfl.bluebrain.nexus.kg.TestHelper
+import ch.epfl.bluebrain.nexus.kg.config.AppConfig._
+import ch.epfl.bluebrain.nexus.kg.resources.{Id, ProjectRef}
+import ch.epfl.bluebrain.nexus.kg.resources.file.File.{Digest, FileAttributes, FileDescription}
+import ch.epfl.bluebrain.nexus.kg.resources.syntax._
+import ch.epfl.bluebrain.nexus.kg.storage.Storage.ExternalDiskStorage
+import ch.epfl.bluebrain.nexus.storage.client.StorageClient
+import org.mockito.{IdiomaticMockito, Mockito}
+import org.scalatest._
+import ch.epfl.bluebrain.nexus.storage.client.types.FileAttributes.{Digest => StorageDigest}
+import ch.epfl.bluebrain.nexus.storage.client.types.{FileAttributes => StorageFileAttributes}
+
+class ExternalDiskStorageOperationsSpec
+    extends ActorSystemFixture("ExternalDiskStorageOperationsSpec")
+    with WordSpecLike
+    with Matchers
+    with BeforeAndAfter
+    with IdiomaticMockito
+    with IOEitherValues
+    with Randomness
+    with Resources
+    with TestHelper {
+
+  private val endpoint = "http://nexus.example.com/v1"
+
+  private implicit val sc: StorageConfig = StorageConfig(
+    DiskStorageConfig(Paths.get("/tmp"), "SHA-256", read, write),
+    ExternalDiskStorageConfig("http://example.com", None, "SHA-256", read, write),
+    S3StorageConfig("MD5", read, write),
+    "password",
+    "salt"
+  )
+
+  sealed trait Ctx {
+    val cred                              = genString()
+    implicit val token: Option[AuthToken] = Some(AuthToken(cred))
+    val path                              = Uri.Path(s"${genString()}/${genString()}")
+    // format: off
+    val storage = ExternalDiskStorage(ProjectRef(genUUID), genIri, 1L, false, false, "SHA-256", endpoint, Some(cred.encrypt), genString(), Permission.unsafe(genString()), Permission.unsafe(genString()))
+    val attributes = FileAttributes(s"$endpoint/${storage.folder}/$path", path, s"${genString()}.json", `application/json`, 12L, Digest("SHA-256", genString()))
+    // format: on
+  }
+
+  private def sourceInChunks(input: String): AkkaSource =
+    Source.fromIterator(() ⇒ input.grouped(10000).map(ByteString(_)))
```

Comment:
Unicode arrow

Reply:
I wasn't aware that unicode arrows were a problem. If I "fix" this, I do it everywhere

Reply:
It's bad practice since ligatures are supported by IDEs, we don't really see it anywhere outside of Akka's codebase.